app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt

```diff
@@ -165,14 +162,41 @@ class MoveFragment : MenuFoldersFragment() {
         }
     }
 
-    private fun toggleFolderListsVisibility(isSearching: Boolean) = with(binding) {
-        searchResultsList.isVisible = isSearching
-        moveFoldersLists.isGone = isSearching
+    private fun toggleFolderListsVisibility(query: String?, allFolders: List<Folder>) {
+        isSearching = !query.isNullOrBlank()
+        if (!isSearching) folderAdapter.setFolders(allFolders, moveViewModel.currentFolderId)
     }
 
-    private fun observeSearchResults() {
-        moveViewModel.filterResults.observe(viewLifecycleOwner) { folders ->
-            searchResultsAdapter.setFolders(folders, currentFolderId)
+    private fun observeSearchResults() = with(moveViewModel) {
+        filterResults.observe(viewLifecycleOwner) { folders ->
+            if (isSearching) folderAdapter.setFolders(folders, currentFolderId)
         }
     }
+
+    private fun observeNewFolderCreation() = with(mainViewModel) {
+        newFolderResultTrigger.observe(viewLifecycleOwner) { inputDialog.resetLoadingAndDismiss() }
+        isMovedToNewFolder.observe(viewLifecycleOwner) { isFolderCreated ->
+            if (isFolderCreated) findNavController().popBackStack()
+        }
+    }
+
+    private fun onFolderSelected(folderId: String): Unit = with(navigationArgs) {
+        mainViewModel.moveThreadsOrMessageTo(folderId, threadsUids.toList(), messageUid)
+        findNavController().popBackStack()
+    }
+
+    /**
+     * Asynchronously validate folder name locally
+     * @return error string, otherwise null
+     */
+    private fun checkForFolderCreationErrors(folderName: CharSequence): String? = when {
+        folderName.length > 255 -> getString(R.string.errorNewFolderNameTooLong)
+        folderName.contains(Regex(INVALID_CHARACTERS_PATTERN)) -> getString(R.string.errorNewFolderInvalidCharacter)
+        folderController.getRootFolder(folderName.toString()) != null -> context?.getString(R.string.errorNewFolderAlreadyExists)
+        else -> null
+    }
```

Comment:
We should put some of this code in common with the menu drawer fragment. This is a good example but this might not be the only one, maybe there's more logic around the new folder creation

Reply:
Done in https://github.com/Infomaniak/android-kMail/pull/1899

Reply:
Same here but as in a following comment but, is there a is there a specific reason to delay the fix to a future PR instead of including it in this one?

Reply:
Both PRs have been merged into only 1 PR.